src/components/lib/LanguageDropdown.js

```diff
@@ -2,13 +2,13 @@ import React from "react"
 import { StaticQuery, graphql } from "gatsby"
 import PropTypes from "prop-types"
 import classNames from "classnames"
-import { useLocation } from "@reach/router"
 
 import Dropdown from "./Dropdown"
 import { CustomLocaleLink as Link } from "../LocaleLink"
 
 const LanguageDropdownTemplate = ({ languages = [], selectedLanguage }) => {
-  const location = useLocation()
+  const location =
+    typeof window !== 'undefined' && window.location || { pathname: "" }
```

Comment:
Could we try/catch `useLocation` and fall back on `window.location` if it throws? Wondering if that's cleaner or not, but it feels like it would be.

Reply:
Will try it

Reply:
(Initial reaction is I'm worried it won't work bc of block scoping)

Reply:
Short version of what I'm thinking here is basically this monster:

```
  let location
  try {
    location = useLocation()
  } catch (_) {
    location = typeof ...
  }
  ...
```

Reply:
I tried it this way but it yielded a compilation error:
```
/Users/erinng/Dev/tbtc-website/src/components/lib/LanguageDropdown.js
  14:16  error  React Hook "useLocation" is called conditionally. React Hooks must be called in the exact same order in every component render  react-hooks/rules-of-hooks
```

Just pushed an alternative. Luckily since `window` is in the catch block we didn't need to use the `typeof` check to pass the preview build (at least when I tried it locally)